pkg/cloudprovider/provider/kubevirt/provider.go

```diff
@@ -73,6 +73,10 @@ const (
 	machineDeploymentLabelKey = "md"
 	// httpSource defines the http source type for VM Disk Image.
 	httpSource imageSource = "http"
+	// dockerSource defines the docker source type for VM Disk Image.
+	dockerSource imageSource = cdiv1beta1.RegistrySchemeDocker
+	// ociArchiveSource defines the oci-archive source type for VM Disk Image.
+	ociArchiveSource imageSource = cdiv1beta1.RegistrySchemeOci
```

Comment:
why do we need to have distinction between docker and OCI? Docker is OCI compliant. I would basically leave this to CDI to recognize the repo.

I would prefer if we can introduce a new imageSource type `registry`

Reply:
this originates from [the upstream](https://github.com/kubevirt/containerized-data-importer/blob/v1.55.2/staging/src/kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1/types.go#L174-L177). It was simple enough to pass through both to CDI as implemented there.

Reply:
I think that's just implementation detail we don't need to know. 
We are mostly concerned about the `source` type, like `registry`.

https://github.com/kubevirt/containerized-data-importer/blob/main/manifests/example/registry-image-block-datavolume.yaml#L8

That's why I suggest that we only introduce `registry` type.
The same as we did for:
* http https://github.com/kubevirt/containerized-data-importer/blob/main/manifests/example/vm-dv.yaml#L22
* pvc https://github.com/kubevirt/containerized-data-importer/blob/main/manifests/example/clone-datavolume.yaml#L8